apps/api/src/services/library-traffic.ts

```diff
@@ -8,27 +8,23 @@ type LibraryTrafficServiceInput = z.infer<typeof libraryTrafficQuerySchema>;
 
 type LibraryTrafficServiceOutput = z.infer<typeof libraryTrafficSchema>;
 
-const transformTrafficRow = (
-  row: typeof libraryTraffic.$inferSelect,
-): LibraryTrafficServiceOutput => ({
-  id: row.id,
-  locationName: row.locationName,
-  trafficCount: row.trafficCount,
-  trafficPercentage: Number(row.trafficPercentage),
-  timestamp: row.timestamp.toISOString(),
-});
-
 export class LibraryTrafficService {
   constructor(private readonly db: ReturnType<typeof database>) {}
 
   async getLibraryTraffic(
     input: LibraryTrafficServiceInput,
   ): Promise<LibraryTrafficServiceOutput[]> {
     const rows = await this.db
-      .select()
+      .select({
+        id: libraryTraffic.id,
+        locationName: libraryTraffic.locationName,
+        trafficCount: libraryTraffic.trafficCount,
+        trafficPercentage: libraryTraffic.trafficPercentage,
+        timestamp: libraryTraffic.timestamp,
+      })
       .from(libraryTraffic)
       .where(input.locationName ? eq(libraryTraffic.locationName, input.locationName) : undefined);
 
-    return rows.map(transformTrafficRow);
+    return rows.map((row) => libraryTrafficSchema.parse(row));
```

Comment:
two things:
1. by convention, the REST handler and GraphQL resolver should call `.parse` on some schema to validate what they get back from the service
2. you should redefine `libraryTrafficSchema` as an array of some other object (say, `libraryTrafficEntry`), and parse that single schema against the returned array of row objects